### ISSUE TITLE
v1.3.0 makes illegal move

### ISSUE THREAD

**Issue by tissatussa** [sictransit/woodpusher#34]
During many 5m+3s games of v1.3.0 against other engines, Woodpusher sometimes makes an illegal move. Same as v1.2.1 .. It can happen at the second move or much later in a game. I gathered 4 instances :

![woodpusher-v1 3 0-illegal-move-vs-Snowy](https://github.com/user-attachments/assets/f092157f-d90b-4e16-a5b0-42d57d9f0064)

![woodpusher-v1 3 0-illegal-move-vs-BACE](https://github.com/user-attachments/assets/2d2ca67e-8dfb-4609-84d9-8d0094520eeb)

![woodpusher-v1 3 0-illegal-move-vs-Victoire](https://github.com/user-attachments/assets/16c51ec7-5241-4438-8e33-90160315d8fc)

![woodpusher-v1 3 0-illegal-move-vs-TSCP](https://github.com/user-attachments/assets/1b017fda-c54b-4d52-8c6b-bea103515ae9)

Here are the ZIPped logs : [v1.3.0-illegal-move-logs.zip](https://github.com/user-attachments/files/17969285/v1.3.0-illegal-move-logs.zip)
@ the 4 CuteChess engine debugs and one big file written by the engine.

Making an illegal move seems to appear randomly, once in 10 games or so.
I can't see a pattern to find a cause, do you ? HTH

**Comment by tissatussa** [comment]
it's a pitty :

![woodpusher-v1 3 0-illegal-move-vs-KingSlayer](https://github.com/user-attachments/assets/abca6d42-021f-4fb7-a6f5-fd8e81cd7d10)

**Comment by sictransit** [comment]
I think I nailed it down to the async processing of UCI commands. In the log you provided it was apparent that the `position` command was sent just 1 ms before a `go`. That made me think that the engine wasn't updated with the latest position. It seemed likely as the illegal moves always were a repetition of the previous one. 

I've now removed all async processing for UCI commands, except `go` of course. That command sets a busy flag making the engine ignore everything during processing except a `quit` or a `stop`.

I've made the above changes in the same branch as a new opening book and bumped the version to 1.4.0.

Your logs were really helpful. Thank you!

(BTW, it seems my machine is about 3x as fast as yours when I compare `nps`. That explains why I see a higher ELO.)
